perceval/simulators/simulator.py

```diff
@@ -501,8 +534,15 @@ def evolve_density_matrix(self, dm: DensityMatrix) -> DensityMatrix:
                               u_evolve_indices,
                               u_evolve_indptr),
                              shape=dm.shape)
+        return u_evolve
 
-        inter_matrix = u_evolve @ extract_upper_triangle(dm.mat) @ u_evolve.T.conj()
-        out_matrix = inter_matrix + inter_matrix.T.conjugate(copy=False)
-
-        return DensityMatrix(out_matrix, index=dm.index)
+    @staticmethod
+    def _get_density_matrix_input_list(dm: DensityMatrix) -> list:
+        """
+        get the input_list necessary to density_matrix evolution
```

Comment:
`input_list` is used in both evolve and probs of density matrix. Perhaps a different way of documenting/classifying it?